Gemfile

```diff
@@ -0,0 +1,27 @@
+#
+# SIMP_GEM_SERVERS   | a space/comma delimited list of rubygem servers
+# PUPPET_VERSION     | specifies the version of the puppet gem to load
+# FACTER_GEM_VERSION | specifies the version of the facter to load
+puppetversion = ENV.fetch('PUPPET_VERSION', '~> 5.5')
+gem_sources   = ENV.key?('SIMP_GEM_SERVERS') ? ENV['SIMP_GEM_SERVERS'].split(/[, ]+/) : ['https://rubygems.org']
+
+gem_sources.each { |gem_source| source gem_source }
+
+gemspec
+
+gem 'puppet', puppetversion, :require => false
+gem 'bundler'
+gem 'rake'
+
+if facterversion = ENV['FACTER_GEM_VERSION']
+  gem 'facter', facterversion, :require => false
+end
+
+group :test do
+  gem 'rspec'
```

Comment:
`rspec` is already included by the .gemspec file.  Unless the Gemfile's `gem` statement refines a gemspec's gem (like  how `PUPPET_VERSION` refines the `puppet` gem) there shouldn't be any redundant gems.